server/lib/tuist_web/controllers/api/cache/key_value_controller.ex

```diff
@@ -0,0 +1,192 @@
+        Error
+      },
+      not_found: {"No entries found for the given CAS ID", "application/json", Error}
+    }
+  )
+
+  def get_value(%{assigns: %{selected_project: project}} = conn, %{cas_id: cas_id} = _params) do
+    entries = Cache.get_entries_by_cas_id_and_project_id(cas_id, project.id)
+
+    case entries do
+      [] ->
+        conn
+        |> put_status(:not_found)
+        |> json(%Error{message: "No entries found for CAS ID #{cas_id}."})
+
+      _ ->
+        formatted_entries = Enum.map(entries, fn entry -> %{"value" => entry.value} end)
+
+        conn
+        |> put_status(:ok)
+        |> json(%{"entries" => formatted_entries})
+    end
+  end
+
+  operation(:put_value,
+    summary: "Store cache key value entries.",
+    operation_id: "putCacheValue",
+    parameters: [
+      account_handle: [
+        in: :query,
+        type: :string,
+        required: true,
+        description: "The handle of the project's account."
+      ],
+      project_handle: [
+        in: :query,
+        type: :string,
+        required: true,
+        description: "The handle of the project."
+      ]
+    ],
+    request_body: {
+      "CAS entries with CAS ID",
+      "application/json",
+      %Schema{
+        type: :object,
+        properties: %{
+          cas_id: %Schema{type: :string, description: "The CAS identifier"},
+          entries: %Schema{
+            description: "Map of entry keys to encoded values",
+            type: :array,
+            items: %Schema{
+              type: :object,
+              properties: %{
+                value: %Schema{type: :string, description: "The value of the entry"}
+              },
+              required: [:value]
+            }
+          }
+        },
+        required: [:cas_id, :entries]
+      },
+      required: true
+    },
+    responses: %{
+      ok: {
+        "Value stored successfully",
+        "application/json",
+        %Schema{
+          type: :object,
+          properties: %{
+            entries: %Schema{
+              type: :array,
+              items: %Schema{
+                type: :object,
+                properties: %{
+                  id: %Schema{type: :string, description: "The ID of the entry"}
+                },
+                required: [:id]
+              }
+            }
+          },
+          required: [:entries]
+        }
+      },
+      unauthorized: {
+        "You need to be authenticated to access this resource",
+        "application/json",
+        Error
+      },
+      forbidden: {
+        "The authenticated subject is not authorized to perform this action",
+        "application/json",
+        Error
+      },
+      not_found: {"The project was not found", "application/json", Error}
+    }
+  )
+
+  def put_value(
+        %{assigns: %{selected_project: project}, body_params: %{entries: entries, cas_id: cas_id}} = conn,
+        _params
+      ) do
+    inserted_entries =
+      Enum.map(entries, fn entry ->
```

Comment:
usually, there's just a single entry per cas_id. If we see the list growing, we should probably insert this as a batch.